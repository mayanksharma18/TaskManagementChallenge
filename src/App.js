import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import Home from "./components/Home";
import "./App.css";
import BoardPage from "./components/Board/BoardPage/BoardPage";
import { AppStateProvider } from "../src/contexts/AppStateContext";

function App() {
  return (
    <div className="App">
      <AppStateProvider>
        <Router>
          <Routes>
            <Route exact path="/" element={<Home />} />
            <Route exact path="/board/:boardName" element={<BoardPage />} />
          </Routes>
        </Router>
      </AppStateProvider>
    </div>
  );
}

export default App;
