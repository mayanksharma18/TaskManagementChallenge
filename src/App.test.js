import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";

test("renders Home component on / route", () => {
  render(<App />);
  expect(screen.getByText('Task Management App')).toBeInTheDocument();
  expect(screen.getByText('Create board')).toBeInTheDocument();
  expect(screen.getByText('Task Management App')).toBeInTheDocument();
  expect(render(<App />).container).toMatchSnapshot();
});
