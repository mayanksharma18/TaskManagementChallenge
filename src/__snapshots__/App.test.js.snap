// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home component on / route 1`] = `
<div>
  <div
    class="App"
  >
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
          >
            Task Management App
          </div>
        </div>
      </header>
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-y44dvl-MuiToolbar-root"
      >
        <div
          style="display: flex; flex-direction: column; margin-top: 20px;"
        >
          <div>
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1mvo7s5-MuiPaper-root-MuiCard-root"
            >
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1l9sosn-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Create board
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            style="margin-top: 80px;"
          >
            <div>
              <h5
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
              >
                Current boards:
              </h5>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                No boards available.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
